### ISSUE TITLE
[SUGGESTION]: Adding a "mark unread" user toggle on NPCs.

### ISSUE THREAD

**Issue by IdlerDude** [pokeclicker/pokeclicker#3761]
### Current Behavior

This suggestion involves a small but new game function.

### Improved Behavior

Add a "mark unread" or "mark undone" toggle on NPCs... this is a way for the player to keep track of what NPCs they still have to interact with later because they are too focused on some other task right now. We can even add a Map Legend color for it so that the player can go back to any previous unfulfilled NPC hint they missed. BUT it's all user directed. It can be more integrated into game mechanics but that also increases code complexity. Options are to initialize them to blank OR automatically set, depending how the community sees best. 

### Reason for change

The great thing about questlines, they're always available and it's clear when they've ended. So this kind of provides a similar function. As I hinted above, keeping track of all the different in game "objectives" is demanding. The achievements and questlines do not do this. This could be a mechanic to fill that gap?

This will not help with those players that do not read, but it would provide an extra hint to them, one which could be reinforced in chat?

### Detailed Description

It would be nice to see this idea get fleshed out with more feedback.

**Comment by DraKay** [comment]
And add an icon over the city similar to Google maps pin? So you can see city color and missing npc at same time?

**Comment by RegisCoaxans** [comment]
What if the location just turns « incomplete » again ? Pins on the map might be a mess.

**Comment by Qwertypop04** [comment]
Perhaps instead of effecting the map, read NPCs should have a different colour, similar to cleared gyms

**Comment by IdlerDude** [comment]
Giving read NPCs a different color is compatible with the original suggestion. I just may not have explicitly stated it that way.

**Comment by Aegyo** [comment]
Would #3576 take care of this then, if it was visible on the map too?

**Comment by IdlerDude** [comment]
#3576 is definitely related to this BUT goes in a different direction. the solution it proposes is not as good... you would still have to check every location for NPCs, rather than letting the Map Legend do that work for you.. once the location is determined then what is proposed there would be a great finishing touch.

If you prefer merging these two issues, that is fine, but only as long as the Map Legend aspect is considered. If not, then these are separate suggestions.

**Comment by Aegyo** [comment]
Yep, I agree, it only helps with this issue if it the npc read status is visible from the map

**Comment by RegisCoaxans** [comment]
That kind of is what I was designing. I think these two issues must result in one proper solution.
*Not read yet* NPC should not interfer with the map, but a *not done* status should because the user explicitly think that the location is uncompleted for them. Although, this set status should definitely count as *read* for a theoretical future NPCRequirement (otherwise, the player would mess with the requirements), thus requires to be a different status than *not read yet*.

**Comment by Aegyo** [comment]
What reason would someone have for marking an NPC as unread after they have read it?

**Comment by RegisCoaxans** [comment]
That is the point of this issue. Is it not ? At least, that is how I understood it...

**Comment by Aegyo** [comment]
Yes that is what the issue asks for, I'm trying to find out why you would want specifically that vs just knowing when there is an NPC you haven't ever read the text of

**Comment by RegisCoaxans** [comment]
Well, it is up to Idler. Which is explained in **improved behavior**.

**Comment by jk13pclick** [comment]
This is approved, with the caveat of it not being a map filter. We don't want to see that legend/menu cluttered.
